File: platforms/web/examples/scene/light/set_range.js
import { Light } from "fragmentcolor";

const lamp = Light.point([0.0, 2.0, 0.0], [1.0, 1.0, 1.0]);
lamp.setRange(8.0);
const negative = lamp.setRange(-1.0);

// Directional lights have no range — the call errors.
const sun = Light.directional([0.0, -1.0, 0.0], [1.0, 1.0, 1.0]);
const unsupported = sun.setRange(5.0);

File: platforms/swift/examples/scene/light/set_range.swift
import FragmentColor

let lamp = Light.point([0.0, 2.0, 0.0], [1.0, 1.0, 1.0])
lamp.setRange(8.0)
let negative = lamp.setRange(-1.0)

// Directional lights have no range — the call errors.
let sun = Light.directional([0.0, -1.0, 0.0], [1.0, 1.0, 1.0])
let unsupported = sun.setRange(5.0)